src/linear_algebra/matrix/reindex.lean

```diff
@@ -52,14 +57,52 @@ rfl
   reindex_linear_equiv (equiv.refl m) (equiv.refl n) = linear_equiv.refl R _ :=
 linear_equiv.ext $ λ _, rfl
 
-lemma reindex_linear_equiv_mul {o o' : Type*} [fintype o] [fintype o'] [semiring R]
+lemma reindex_linear_equiv_trans [semiring R] (e₁ : m ≃ m') (e₂ : n ≃ n') (e₁' : m' ≃ m'')
+  (e₂' : n' ≃ n'') : (reindex_linear_equiv e₁ e₂).trans (reindex_linear_equiv e₁' e₂') =
+   @reindex_linear_equiv _ _ _ _ _ _ _ _ R _ (e₁.trans e₁') (e₂.trans e₂') :=
+by { ext, refl }
+
+lemma reindex_linear_equiv_comp [semiring R] (e₁ : m ≃ m') (e₂ : n ≃ n') (e₁' : m' ≃ m'')
+  (e₂' : n' ≃ n'') : (@reindex_linear_equiv _ _ _ _ _ _ _ _ R _ e₁' e₂') ∘
+   (@reindex_linear_equiv _ _ _ _ _ _ _ _ R _ e₁ e₂)
+  = (reindex_linear_equiv (e₁.trans e₁') (e₂.trans e₂')) :=
+by { rw [← reindex_linear_equiv_trans], refl }
+
+lemma reindex_linear_equiv_comp_apply [semiring R] (e₁ : m ≃ m') (e₂ : n ≃ n') (e₁' : m' ≃ m'')
+  (e₂' : n' ≃ n'') (M : matrix m n R) :
+  (reindex_linear_equiv e₁' e₂') (reindex_linear_equiv e₁ e₂ M) =
+    reindex_linear_equiv (e₁.trans e₁') (e₂.trans e₂') M :=
+by rw [← reindex_linear_equiv_comp e₁ e₂ e₁' e₂']
+
+lemma reindex_linear_equiv_one [semiring R] [decidable_eq m] [decidable_eq m']
+  (e : m ≃ m') : (reindex_linear_equiv e e (1 : matrix m m R)) = 1 :=
+minor_one_equiv e.symm
+
+variables {o o' : Type*} [fintype o] [fintype o']
+
+lemma reindex_linear_equiv_mul [semiring R]
   (eₘ : m ≃ m') (eₙ : n ≃ n') (eₒ : o ≃ o') (M : matrix m n R) (N : matrix n o R) :
   reindex_linear_equiv eₘ eₒ (M ⬝ N) =
     reindex_linear_equiv eₘ eₙ M ⬝ reindex_linear_equiv eₙ eₒ N :=
 minor_mul_equiv M N _ _ _
 
-/-- For square matrices, the natural map that reindexes a matrix's rows and columns with equivalent
-types, `matrix.reindex`, is an equivalence of algebras. -/
+lemma mul_reindex_linear_equiv_mul_one [semiring R] [decidable_eq o] (e₁ : o ≃ n) (e₂ : o ≃ n')
+  (M : matrix m n R) : M.mul (reindex_linear_equiv e₁ e₂ 1) =
+    reindex_linear_equiv (equiv.refl m) (e₁.symm.trans e₂) M :=
```

Comment:
Fixed a typo:
```suggestion
lemma mul_reindex_linear_equiv_one [semiring R] [decidable_eq o] (e₁ : o ≃ n) (e₂ : o ≃ n')
  (M : matrix m n R) : M.mul (reindex_linear_equiv e₁ e₂ 1) =
    reindex_linear_equiv (equiv.refl m) (e₁.symm.trans e₂) M :=
```

Reply:
Following the guidelines I described, you might want to define a lemma:
```lean
lemma mul_minor_one [semiring R] [decidable_eq o] (e₁ : n ≃ o) (e₂ : n' ≃o')
  (M : matrix m n R) : M.mul (minor e₁ e₂ 1) = minor id (e₂.trans e₁.symm) M :=
sorry
```
However, this lemma is itself already quite specific, so I'd be happy to accept this PR without the `minor` version.

Reply:
I will implement this.

Reply:
I have added a lemma `mul_minor_one`: is it the one you were looking for? If yes, I'll go on and prove it. I have also modified the `doc` a bit.